src/main/java/no/nav/pto/veilarbportefolje/persononinfo/PdlRepository.java

```diff
@@ -0,0 +1,77 @@
+package no.nav.pto.veilarbportefolje.persononinfo;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import no.nav.common.types.identer.AktorId;
+import no.nav.pto.veilarbportefolje.persononinfo.PdlResponses.PDLIdent;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static no.nav.pto.veilarbportefolje.postgres.PostgresUtils.queryForObjectOrNull;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class PdlRepository {
+    @Qualifier("PostgresJdbc")
+    private final JdbcTemplate db;
+
+    @Transactional
+    public void upsertIdenter(List<PDLIdent> pdlIdenter) {
+        List<String> identer = pdlIdenter.stream().map(PDLIdent::getIdent).toList();
+        slettIdenter(identer);
+
+        String nyLokalIdent = db.queryForObject("select nextval('PDL_PERSON_SEQ')", String.class);
+        pdlIdenter.forEach(ident -> insertIdent(nyLokalIdent, ident));
+    }
+
+    @Transactional
+    public void slettLokalIdentlagringHvisIkkeUnderOppfolging(AktorId aktorId) {
+        String lokalIdent = hentLokalIdent(aktorId.get());
+        if (harIdentUnderOppfolging(lokalIdent)) {
+            log.info("""
+                            Sletter ikke identer lagret på aktorId: {}.
+                            Da en eller flere relaterte identer på lokalIdent: {} er under oppfolging.
+                            """,
+                    aktorId, lokalIdent);
+            return;
+        }
+        log.info("Sletter identer lagret på aktorId: {}, lokalIdent: {}.", aktorId, lokalIdent);
+        slettLagreteIdenter(lokalIdent);
+    }
+
+    public String hentLokalIdent(String lookUpIdent) {
```

Comment:
Hadde kanskje vært fint med en egen type for lokalIdent? F.eks. `LokalPersonIdent`?

Reply:
Kanskje, men så har jeg ikke lyst til at denne identen blir brukt uten for klassen, kanskje det er ok med en intern klasse?

Reply:
Enig i det, den bør være privat inner class i tilfelle